tests/console/ansible.pm

```diff
@@ -0,0 +1,125 @@
+
+    # Learn public SSH host keys
+    assert_script_run 'ssh-keyscan localhost >> ~/.ssh/known_hosts';
+
+    # Check that we can connect to localhost via SSH
+    validate_script_output('ssh localhost whoami', sub { m/$testapi::username/ });
+
+    my $hostname = script_output 'hostname';
+    validate_script_output 'ansible -m setup localhost | grep ansible_hostname', sub { m/$hostname/ };
+
+    my $arch = get_var 'ARCH';
+    validate_script_output 'ansible -m setup localhost | grep ansible_architecture', sub { m/$arch/ };
+
+    # Download data/console/ansible/ directory
+    assert_script_run 'curl ' . data_url('console/ansible/') . ' | cpio -id';
+
+    # In order to use ansible-test we need to follow this directory tree 
+    assert_script_run "mkdir -p ~/ansible_collections/openqa";
+    assert_script_run "mv data ~/ansible_collections/openqa/ansible";
+    assert_script_run 'cd ~/ansible_collections/openqa/ansible';
+
+    # Download base folder files
+    #assert_script_run 'curl -so main.yaml ' . data_url 'console/ansible/playbook_main.yaml';
+    #assert_script_run 'curl -so hosts ' . data_url 'console/ansible/playbook_hosts';
+
+    # Download role specific files
+    #assert_script_run 'curl -so roles/test/tasks/main.yaml ' . data_url 'console/ansible/role_task.yaml';
+    #assert_script_run 'curl -so roles/test/files/static.txt ' . data_url 'console/ansible/role_file.txt';
+    #assert_script_run 'curl -so roles/test/handlers/main.yaml ' . data_url 'console/ansible/role_handler.yaml';
+    #assert_script_run 'curl -so roles/test/templates/README.txt.j2 ' . data_url 'console/ansible/role_template.txt.j2';
+
+    # Download community.general modules because of zypper module
+    assert_script_run 'ansible-galaxy collection install community.general';
+
+    # Check the playbook
+    assert_script_run 'ansible-playbook -i hosts main.yaml --check';
+
+    # Run the ansible sanity test
+    assert_script_run 'ansible-test sanity';
+
+    # Run the playbook
+    assert_script_run 'ansible-playbook -i hosts main.yaml';
+
+    # Test that /tmp/ansible/uname.txt created by ansible has desired content
+    my $uname = script_output 'uname -r';
+    validate_script_output 'cat /tmp/ansible/uname.txt', sub { m/$uname/ };
+
+    # Test that /tmp/ansible/static.txt contains the static file
+    validate_script_output 'cat /tmp/ansible/static.txt', sub {m/stay the same/ };
+
+    # Validate that /tmp/ansible/os-release points to /etc/os-release
+    validate_script_output 'readlink /tmp/ansible/os-release', sub { m/\/etc\/os-release/ };
+
+    # Test that /home/johnd/README.txt is readable and contains the expanded template
+    validate_script_output 'sudo -u johnd cat /home/johnd/README.txt', sub { m/my $arch dynamic kingdom/ };
+
+    # Check that Ed - the command line text edit is installed
+    assert_script_run 'which ed';
+}
+
+sub cleanup {
+    assert_script_run 'cd';
```

Comment:
`assert_script_run` when executed in the `post_fail_hook` might have unintended side-effects (`die` while cleanup) - I should use `script_run` instead, which doesn't has this disadvantage.

Reply:
If I remember correctly @okurz had opinion about this but I kinda forgot it. Oliver can you help? :smile:

Reply:
Actually why not :-) Let's also test that it is properly uninstallable :-)